### PR TITLE
Update to GHC 8.6.4

### LINKED ISSUE #1415: Does this compile with GHC 8.6?

### PR DESCRIPTION
**Pull request by ranjitjhala**
Fixes #1415 

- [x] fix a bunch of regression tests, 
- [x] deal with Desugar/SourceSpan/SourceNote ... (c.f. `ElabLocation.hs` test)

### CODE REVIEW

**Comment by ranjitjhala** [comment]
@nikivazou or @atondwal this looks like its ready to merge. 

Any reason we shouldn't do it? [will update README etc.]

**Comment by spacekitteh** [comment]
Why remove machine floats etc from mkLit?

File: CHANGES.md [hunk 1]
```diff
@@ -2,9 +2,17 @@
 
 ## NEXT 
 
+## 0.8.6.0
+
+- Support for GHC 8.6.4 
+- New structural termination checker (used by default)
+- Miscellaneous bug fixes for PLE 
+- Need to run LH on imported libs (with source) first; can use `--compile-spec` to avoid checking. 
+
```
Comment (nikivazou) [inline CHANGES.md:12]: the type class laws are also in, correct?

**Review by nikivazou — COMMENTED** [review]
So, we are directly using ghc's desugarer and we do not need to duplicated from now on?

File: CHANGES.md (hunk 1 above)
Comment (ranjitjhala) [reply CHANGES.md:12]: @nikivazou  --

1. Oops yes, I will update that in the `CHANGES.md`
2. That's correct, I removed the `Desugarer` stuff as GHC's `SourceNote`s _appear_ to be working properly w.r.t. preserving source-span info for error reporting and type annotations too.
@gridaphobe is a bit skeptical, so we may have to revisit this decision later.

@spacekitteh -- 
Were you referring to this `mkLit`? https://github.com/ucsd-progsys/liquidhaskell/pull/1426/files#diff-9b51e9aee825d104afb416c55069387eL489

If so, note that the `MachFloat` is unchanged, I just updated the code to reflect how GHC-8.6
represents integer-literals by factoring them into a separate  `LitNumber` constructor.

```
mkLit (LitNumber _ n _) = mkI n
-- mkLit (MachInt64  n)    = mkI n
-- mkLit (MachWord   n)    = mkI n
-- mkLit (MachWord64 n)    = mkI n
-- mkLit (LitInteger n _)  = mkI n
```

Comment (nikivazou) [reply CHANGES.md:12]: It's great that you finally removed the Desugarer! 
-- I did not believe it so, did check few negative checks for location reporting and seems to be correct! So, on my side, let's merge!

**Comment by ranjitjhala** [comment]
Yes, I added a few source-span tests to the error-messages regressions
too...

**Review by nikivazou — APPROVED** [review]

**Pull request merged**